ui/src/app/pages/login/account-creation/account-creation-institution.tsx

```diff
@@ -0,0 +1,330 @@
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+import * as validate from 'validate.js';
+
+import {Button} from 'app/components/buttons';
+import {FlexColumn, FlexRow} from 'app/components/flex';
+import {FormSection} from 'app/components/forms';
+import {Error as ErrorDiv, TextInput} from 'app/components/inputs';
+import {TooltipTrigger} from 'app/components/popups';
+import {SpinnerOverlay} from 'app/components/spinners';
+import {AccountCreationOptions} from 'app/pages/login/account-creation/account-creation-options';
+import {WhyWillSomeInformationBePublic} from 'app/pages/login/account-creation/common-content';
+import {commonStyles} from 'app/pages/login/account-creation/common-styles';
+import {institutionApi} from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import {isBlank, reactStyles} from 'app/utils';
+import {reportError} from 'app/utils/errors';
+import {
+  InstitutionalRole,
+  Profile,
+  PublicInstitutionDetails,
+} from 'generated/fetch';
+import {Dropdown} from 'primereact/dropdown';
+
+const styles = reactStyles({
+  ...commonStyles,
+  publiclyDisplayedText: {
+    fontSize: 12,
+    fontWeight: 400
+  },
+  sectionInput: {
+    width: '14rem',
+    height: '1.5rem'
+  },
+  text: {
+    fontSize: 14,
+    color: colors.primary,
+    lineHeight: '22px',
+  }
+});
+
+// TODO: this is copy-pasted from account-creation.tsx. Consider factoring this into a true component.
+function TextInputWithLabel(props) {
```

Comment:
I would very strongly push for putting this into a true component, and I think it would be relatively low effort, considering it seems to already be pretty factored out. Would it be possible to do that within this PR?